### DISCUSSION TITLE
Why was this channel automatically force closed?

### DISCUSSION THREAD

**Discussion by DerEwige** [ACINQ/eclair#2385] [Node operators]
I had a channel to a friend of mine and yesterday he wrote me and asked, why I’d force close the channel. I was confused, as I wasn’t even doing anything on the node when the event happened.

I’ve filtered the log so you can only see entries for the relevant peer, could anyone have a look and tell me what exactly happened.
I can see a disconnect at 16:05: “- waited for too long for a revocation to remoteCommitNumber=28851, disconnecting”
Reconnect at 16:06 followed by an error: peer sent error: ascii='internal error' bin=696e7465726e616c206572726f72
Immediately followed by a force close?

[swisslnd.log](https://github.com/ACINQ/eclair/files/9368060/swisslnd.log)

**Comment by t-bast** [comment]
> followed by an error: peer sent error: ascii='internal error' bin=696e7465726e616c206572726f72

That indicates your friend's node is buggy and is responsible for the force-close. They sent you an `error` message, which [according to the specification](https://github.com/lightning/bolts/blob/master/01-messaging.md#the-error-and-warning-messages), requests your node to force-close the channel.

Your node isn't at fault here, it did everything correctly.

**Reply by DerEwige** [reply]
Thanks for taking the time

**Comment by DerEwige** [comment]
@t-bast 

There is a known error on LND that causes it to send "internal error" responses
https://github.com/lightningnetwork/lnd/discussions/6565

This apparently got so bad, that CLN is currently treatung this error as if it only was a warning 
`We treat LND "internal error" as warnings, not force close events (reverts to v0.10.0 behavior).`
https://github.com/ElementsProject/lightning/releases

This bug is quite annoying and causes a lot of force closes on bussy LND channels =(

**Reply by t-bast** [reply]
Yes, I know, but the only right solution is for lnd to fix those bugs...
